// @nowzoo/ngx-highlight-js/esm2015/lib/ngx-highlight-js.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { NgxHighlightJsService } from './ngx-highlight-js.service';
import { HttpClient } from '@angular/common/http';
export class NgxHighlightJsComponent {
    /**
     * @param {?} service
     * @param {?} http
     */
    constructor(service, http) {
        this.service = service;
        this.http = http;
        this.highlightedHTML = '';
        this.error = null;
        this.afterInit = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.onChange();
        this.afterInit = true;
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        if (this.afterInit) {
            this.onChange();
        }
    }
    /**
     * @return {?}
     */
    fetchText() {
        if (!this.url) {
            return Promise.resolve(this.code || '');
        }
        return new Promise((resolve, reject) => {
            this.http.get(this.url, { responseType: 'text' }).subscribe((text) => resolve(text), (err) => reject(err));
        });
    }
    /**
     * @return {?}
     */
    onChange() {
        this.error = null;
        this.isInitializing = true;
        this.fetchText()
            .then((text) => {
            return this.service.highlight(this.lang, text);
        })
            .then((hl) => {
            this.highlightedHTML = hl;
            this.isInitializing = false;
        })
            .catch(e => {
            this.error = e.toString();
            this.isInitializing = false;
        });
    }
}
NgxHighlightJsComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-highlight-js',
                exportAs: `ngxHighlightJs`,
                template: `<pre><code
      [ngClass]="'hljs ' + lang"
      [innerHTML]="highlightedHTML"></code></pre>`,
            },] },
];
/** @nocollapse */
NgxHighlightJsComponent.ctorParameters = () => [
    { type: NgxHighlightJsService },
    { type: HttpClient }
];
NgxHighlightJsComponent.propDecorators = {
    code: [{ type: Input }],
    url: [{ type: Input }],
    lang: [{ type: Input }]
};
function NgxHighlightJsComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxHighlightJsComponent.prototype.code;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.url;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.lang;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.highlightedHTML;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.error;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.isInitializing;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.afterInit;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.service;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.http;
}
