// @nowzoo/ngx-highlight-js/esm2015/lib/ngx-highlight-js-script.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { Scriptload } from '@nowzoo/sheetload';
import { NGX_HIGHLIGHT_JS_URL } from './ngx-highlight-js-options';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-highlight-js-options";
export class NgxHighlightJsScriptService {
    /**
     * @param {?} _jsURL
     */
    constructor(_jsURL) {
        this._jsURL = _jsURL;
        this._hljs = null;
        this._loadedLanguages = new Map();
    }
    /**
     * @return {?}
     */
    loaded() {
        if (this._loadedPromise) {
            return this._loadedPromise;
        }
        this._loadedPromise = new Promise((resolve, reject) => {
            const /** @type {?} */ url = `${this._jsURL}/highlight.min.js`;
            Scriptload.load(url)
                .then(() => {
                setTimeout(() => {
                    this._hljs = window['hljs'];
                    resolve(this._hljs);
                }, 10);
            })
                .catch(reject);
        });
        return this._loadedPromise;
    }
    /**
     * @param {?} lang
     * @return {?}
     */
    loadLanguage(lang) {
        let /** @type {?} */ p = this._loadedLanguages.get(lang);
        if (p) {
            return p;
        }
        if (this._hljs && this._hljs.listLanguages().indexOf(lang) > -1) {
            p = Promise.resolve();
            this._loadedLanguages.set(lang, p);
            return p;
        }
        const /** @type {?} */ url = `${this._jsURL}/languages/${lang}.min.js`;
        p = this.loaded()
            .then(() => {
            return Scriptload.load(url);
        });
        this._loadedLanguages.set(lang, p);
        return p;
    }
}
NgxHighlightJsScriptService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
NgxHighlightJsScriptService.ctorParameters = () => [
    { type: String, decorators: [{ type: Inject, args: [NGX_HIGHLIGHT_JS_URL,] }] }
];
/** @nocollapse */ NgxHighlightJsScriptService.ngInjectableDef = i0.defineInjectable({ factory: function NgxHighlightJsScriptService_Factory() { return new NgxHighlightJsScriptService(i0.inject(i1.NGX_HIGHLIGHT_JS_URL)); }, token: NgxHighlightJsScriptService, providedIn: "root" });
function NgxHighlightJsScriptService_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxHighlightJsScriptService.prototype._loadedLanguages;
    /** @type {?} */
    NgxHighlightJsScriptService.prototype._hljs;
    /** @type {?} */
    NgxHighlightJsScriptService.prototype._loadedPromise;
    /** @type {?} */
    NgxHighlightJsScriptService.prototype._jsURL;
}
