// @nowzoo/ngx-highlight-js/esm2015/lib/ngx-highlight-js.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { NgxHighlightJsThemeService } from './ngx-highlight-js-theme.service';
import { NgxHighlightJsScriptService } from './ngx-highlight-js-script.service';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-highlight-js-theme.service";
import * as i2 from "./ngx-highlight-js-script.service";
export class NgxHighlightJsService {
    /**
     * @param {?} themeService
     * @param {?} scriptService
     */
    constructor(themeService, scriptService) {
        this.themeService = themeService;
        this.scriptService = scriptService;
    }
    /**
     * @return {?}
     */
    get theme() {
        return this.themeService._theme;
    }
    /**
     * @return {?}
     */
    loaded() {
        if (this._loadedPromise) {
            return this._loadedPromise;
        }
        const /** @type {?} */ promises = [this.scriptService.loaded(), this.themeService.initialized()];
        this._loadedPromise = Promise.all(promises)
            .then((results) => {
            this._hljs = results[0];
            return (this._hljs);
        });
        return this._loadedPromise;
    }
    /**
     * @param {?} lang
     * @return {?}
     */
    loadLanguage(lang) {
        return this.scriptService.loadLanguage(lang);
    }
    /**
     * @param {?} name
     * @return {?}
     */
    loadTheme(name) {
        return this.themeService.loadTheme(name);
    }
    /**
     * @param {?} lang
     * @param {?} code
     * @return {?}
     */
    highlight(lang, code) {
        return this.loaded()
            .then(() => {
            return this.loadLanguage(lang);
        })
            .then(() => {
            const /** @type {?} */ hl = this._hljs.highlight(lang, code, true).value;
            return hl;
        });
    }
}
NgxHighlightJsService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
NgxHighlightJsService.ctorParameters = () => [
    { type: NgxHighlightJsThemeService },
    { type: NgxHighlightJsScriptService }
];
/** @nocollapse */ NgxHighlightJsService.ngInjectableDef = i0.defineInjectable({ factory: function NgxHighlightJsService_Factory() { return new NgxHighlightJsService(i0.inject(i1.NgxHighlightJsThemeService), i0.inject(i2.NgxHighlightJsScriptService)); }, token: NgxHighlightJsService, providedIn: "root" });
function NgxHighlightJsService_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxHighlightJsService.prototype._hljs;
    /** @type {?} */
    NgxHighlightJsService.prototype._loadedPromise;
    /** @type {?} */
    NgxHighlightJsService.prototype.themeService;
    /** @type {?} */
    NgxHighlightJsService.prototype.scriptService;
}
