// @nowzoo/ngx-highlight-js/esm5/lib/ngx-highlight-js-theme.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { Sheetload } from '@nowzoo/sheetload';
import { NGX_HIGHLIGHT_JS_DEFAULT_THEME, NGX_HIGHLIGHT_JS_URL } from './ngx-highlight-js-options';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-highlight-js-options";
var NgxHighlightJsThemeService = /** @class */ (function () {
    function NgxHighlightJsThemeService(_defaultTheme, _jsURL) {
        this._defaultTheme = _defaultTheme;
        this._jsURL = _jsURL;
        this._theme = 'default';
        this._initializedPromise = null;
        this._linkEl = null;
        this._theme = _defaultTheme;
    }
    /**
     * @return {?}
     */
    NgxHighlightJsThemeService.prototype.initialized = /**
     * @return {?}
     */
    function () {
        if (!this._initializedPromise) {
            this._initializedPromise = this.loadTheme(this._theme);
        }
        return this._initializedPromise;
    };
    /**
     * @param {?} name
     * @return {?}
     */
    NgxHighlightJsThemeService.prototype.loadTheme = /**
     * @param {?} name
     * @return {?}
     */
    function (name) {
        var _this = this;
        var /** @type {?} */ url = this._jsURL + "/styles/" + name + ".min.css";
        return Sheetload.load(url)
            .then(function (linkEl) {
            if (_this._linkEl) {
                _this._linkEl.remove();
            }
            _this._linkEl = linkEl;
            _this._linkEl.removeAttribute('disabled');
        });
    };
    Object.defineProperty(NgxHighlightJsThemeService.prototype, "theme", {
        get: /**
         * @return {?}
         */
        function () {
            return this._theme;
        },
        enumerable: true,
        configurable: true
    });
    NgxHighlightJsThemeService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    NgxHighlightJsThemeService.ctorParameters = function () { return [
        { type: String, decorators: [{ type: Inject, args: [NGX_HIGHLIGHT_JS_DEFAULT_THEME,] }] },
        { type: String, decorators: [{ type: Inject, args: [NGX_HIGHLIGHT_JS_URL,] }] }
    ]; };
    /** @nocollapse */ NgxHighlightJsThemeService.ngInjectableDef = i0.defineInjectable({ factory: function NgxHighlightJsThemeService_Factory() { return new NgxHighlightJsThemeService(i0.inject(i1.NGX_HIGHLIGHT_JS_DEFAULT_THEME), i0.inject(i1.NGX_HIGHLIGHT_JS_URL)); }, token: NgxHighlightJsThemeService, providedIn: "root" });
    return NgxHighlightJsThemeService;
}());
export { NgxHighlightJsThemeService };
function NgxHighlightJsThemeService_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxHighlightJsThemeService.prototype._theme;
    /** @type {?} */
    NgxHighlightJsThemeService.prototype._initializedPromise;
    /** @type {?} */
    NgxHighlightJsThemeService.prototype._linkEl;
    /** @type {?} */
    NgxHighlightJsThemeService.prototype._defaultTheme;
    /** @type {?} */
    NgxHighlightJsThemeService.prototype._jsURL;
}
