// @nowzoo/ngx-highlight-js/esm5/lib/ngx-highlight-js.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { NgxHighlightJsService } from './ngx-highlight-js.service';
import { HttpClient } from '@angular/common/http';
var NgxHighlightJsComponent = /** @class */ (function () {
    function NgxHighlightJsComponent(service, http) {
        this.service = service;
        this.http = http;
        this.highlightedHTML = '';
        this.error = null;
        this.afterInit = false;
    }
    /**
     * @return {?}
     */
    NgxHighlightJsComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.onChange();
        this.afterInit = true;
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    NgxHighlightJsComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (this.afterInit) {
            this.onChange();
        }
    };
    /**
     * @return {?}
     */
    NgxHighlightJsComponent.prototype.fetchText = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (!this.url) {
            return Promise.resolve(this.code || '');
        }
        return new Promise(function (resolve, reject) {
            _this.http.get(_this.url, { responseType: 'text' }).subscribe(function (text) { return resolve(text); }, function (err) { return reject(err); });
        });
    };
    /**
     * @return {?}
     */
    NgxHighlightJsComponent.prototype.onChange = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.error = null;
        this.isInitializing = true;
        this.fetchText()
            .then(function (text) {
            return _this.service.highlight(_this.lang, text);
        })
            .then(function (hl) {
            _this.highlightedHTML = hl;
            _this.isInitializing = false;
        })
            .catch(function (e) {
            _this.error = e.toString();
            _this.isInitializing = false;
        });
    };
    NgxHighlightJsComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-highlight-js',
                    exportAs: "ngxHighlightJs",
                    template: "<pre><code\n      [ngClass]=\"'hljs ' + lang\"\n      [innerHTML]=\"highlightedHTML\"></code></pre>",
                },] },
    ];
    /** @nocollapse */
    NgxHighlightJsComponent.ctorParameters = function () { return [
        { type: NgxHighlightJsService },
        { type: HttpClient }
    ]; };
    NgxHighlightJsComponent.propDecorators = {
        code: [{ type: Input }],
        url: [{ type: Input }],
        lang: [{ type: Input }]
    };
    return NgxHighlightJsComponent;
}());
export { NgxHighlightJsComponent };
function NgxHighlightJsComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxHighlightJsComponent.prototype.code;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.url;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.lang;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.highlightedHTML;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.error;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.isInitializing;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.afterInit;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.service;
    /** @type {?} */
    NgxHighlightJsComponent.prototype.http;
}
