// @nowzoo/ngx-highlight-js/esm5/lib/ngx-highlight-js.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { NgxHighlightJsThemeService } from './ngx-highlight-js-theme.service';
import { NgxHighlightJsScriptService } from './ngx-highlight-js-script.service';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-highlight-js-theme.service";
import * as i2 from "./ngx-highlight-js-script.service";
var NgxHighlightJsService = /** @class */ (function () {
    function NgxHighlightJsService(themeService, scriptService) {
        this.themeService = themeService;
        this.scriptService = scriptService;
    }
    Object.defineProperty(NgxHighlightJsService.prototype, "theme", {
        get: /**
         * @return {?}
         */
        function () {
            return this.themeService._theme;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    NgxHighlightJsService.prototype.loaded = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this._loadedPromise) {
            return this._loadedPromise;
        }
        var /** @type {?} */ promises = [this.scriptService.loaded(), this.themeService.initialized()];
        this._loadedPromise = Promise.all(promises)
            .then(function (results) {
            _this._hljs = results[0];
            return (_this._hljs);
        });
        return this._loadedPromise;
    };
    /**
     * @param {?} lang
     * @return {?}
     */
    NgxHighlightJsService.prototype.loadLanguage = /**
     * @param {?} lang
     * @return {?}
     */
    function (lang) {
        return this.scriptService.loadLanguage(lang);
    };
    /**
     * @param {?} name
     * @return {?}
     */
    NgxHighlightJsService.prototype.loadTheme = /**
     * @param {?} name
     * @return {?}
     */
    function (name) {
        return this.themeService.loadTheme(name);
    };
    /**
     * @param {?} lang
     * @param {?} code
     * @return {?}
     */
    NgxHighlightJsService.prototype.highlight = /**
     * @param {?} lang
     * @param {?} code
     * @return {?}
     */
    function (lang, code) {
        var _this = this;
        return this.loaded()
            .then(function () {
            return _this.loadLanguage(lang);
        })
            .then(function () {
            var /** @type {?} */ hl = _this._hljs.highlight(lang, code, true).value;
            return hl;
        });
    };
    NgxHighlightJsService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    NgxHighlightJsService.ctorParameters = function () { return [
        { type: NgxHighlightJsThemeService },
        { type: NgxHighlightJsScriptService }
    ]; };
    /** @nocollapse */ NgxHighlightJsService.ngInjectableDef = i0.defineInjectable({ factory: function NgxHighlightJsService_Factory() { return new NgxHighlightJsService(i0.inject(i1.NgxHighlightJsThemeService), i0.inject(i2.NgxHighlightJsScriptService)); }, token: NgxHighlightJsService, providedIn: "root" });
    return NgxHighlightJsService;
}());
export { NgxHighlightJsService };
function NgxHighlightJsService_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxHighlightJsService.prototype._hljs;
    /** @type {?} */
    NgxHighlightJsService.prototype._loadedPromise;
    /** @type {?} */
    NgxHighlightJsService.prototype.themeService;
    /** @type {?} */
    NgxHighlightJsService.prototype.scriptService;
}
