// @nowzoo/ngx-highlight-js/esm5/lib/ngx-highlight-js-script.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { Scriptload } from '@nowzoo/sheetload';
import { NGX_HIGHLIGHT_JS_URL } from './ngx-highlight-js-options';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-highlight-js-options";
var NgxHighlightJsScriptService = /** @class */ (function () {
    function NgxHighlightJsScriptService(_jsURL) {
        this._jsURL = _jsURL;
        this._hljs = null;
        this._loadedLanguages = new Map();
    }
    /**
     * @return {?}
     */
    NgxHighlightJsScriptService.prototype.loaded = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this._loadedPromise) {
            return this._loadedPromise;
        }
        this._loadedPromise = new Promise(function (resolve, reject) {
            var /** @type {?} */ url = _this._jsURL + "/highlight.min.js";
            Scriptload.load(url)
                .then(function () {
                setTimeout(function () {
                    _this._hljs = window['hljs'];
                    resolve(_this._hljs);
                }, 10);
            })
                .catch(reject);
        });
        return this._loadedPromise;
    };
    /**
     * @param {?} lang
     * @return {?}
     */
    NgxHighlightJsScriptService.prototype.loadLanguage = /**
     * @param {?} lang
     * @return {?}
     */
    function (lang) {
        var /** @type {?} */ p = this._loadedLanguages.get(lang);
        if (p) {
            return p;
        }
        if (this._hljs && this._hljs.listLanguages().indexOf(lang) > -1) {
            p = Promise.resolve();
            this._loadedLanguages.set(lang, p);
            return p;
        }
        var /** @type {?} */ url = this._jsURL + "/languages/" + lang + ".min.js";
        p = this.loaded()
            .then(function () {
            return Scriptload.load(url);
        });
        this._loadedLanguages.set(lang, p);
        return p;
    };
    NgxHighlightJsScriptService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    NgxHighlightJsScriptService.ctorParameters = function () { return [
        { type: String, decorators: [{ type: Inject, args: [NGX_HIGHLIGHT_JS_URL,] }] }
    ]; };
    /** @nocollapse */ NgxHighlightJsScriptService.ngInjectableDef = i0.defineInjectable({ factory: function NgxHighlightJsScriptService_Factory() { return new NgxHighlightJsScriptService(i0.inject(i1.NGX_HIGHLIGHT_JS_URL)); }, token: NgxHighlightJsScriptService, providedIn: "root" });
    return NgxHighlightJsScriptService;
}());
export { NgxHighlightJsScriptService };
function NgxHighlightJsScriptService_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxHighlightJsScriptService.prototype._loadedLanguages;
    /** @type {?} */
    NgxHighlightJsScriptService.prototype._hljs;
    /** @type {?} */
    NgxHighlightJsScriptService.prototype._loadedPromise;
    /** @type {?} */
    NgxHighlightJsScriptService.prototype._jsURL;
}
