// @nowzoo/ngx-highlight-js/fesm5/nowzoo-ngx-highlight-js.js
import { InjectionToken, Injectable, Inject, Component, Input, NgModule, defineInjectable, inject } from '@angular/core';
import { Sheetload, Scriptload } from '@nowzoo/sheetload';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var /** @type {?} */ NGX_HIGHLIGHT_JS_URL = new InjectionToken('The base url for highlight.js');
var /** @type {?} */ NGX_HIGHLIGHT_JS_DEFAULT_THEME = new InjectionToken('The default theme for highlight.js');

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxHighlightJsThemeService = /** @class */ (function () {
    function NgxHighlightJsThemeService(_defaultTheme, _jsURL) {
        this._defaultTheme = _defaultTheme;
        this._jsURL = _jsURL;
        this._theme = 'default';
        this._initializedPromise = null;
        this._linkEl = null;
        this._theme = _defaultTheme;
    }
    /**
     * @return {?}
     */
    NgxHighlightJsThemeService.prototype.initialized = /**
     * @return {?}
     */
    function () {
        if (!this._initializedPromise) {
            this._initializedPromise = this.loadTheme(this._theme);
        }
        return this._initializedPromise;
    };
    /**
     * @param {?} name
     * @return {?}
     */
    NgxHighlightJsThemeService.prototype.loadTheme = /**
     * @param {?} name
     * @return {?}
     */
    function (name) {
        var _this = this;
        var /** @type {?} */ url = this._jsURL + "/styles/" + name + ".min.css";
        return Sheetload.load(url)
            .then(function (linkEl) {
            if (_this._linkEl) {
                _this._linkEl.remove();
            }
            _this._linkEl = linkEl;
            _this._linkEl.removeAttribute('disabled');
        });
    };
    Object.defineProperty(NgxHighlightJsThemeService.prototype, "theme", {
        get: /**
         * @return {?}
         */
        function () {
            return this._theme;
        },
        enumerable: true,
        configurable: true
    });
    NgxHighlightJsThemeService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    NgxHighlightJsThemeService.ctorParameters = function () { return [
        { type: String, decorators: [{ type: Inject, args: [NGX_HIGHLIGHT_JS_DEFAULT_THEME,] }] },
        { type: String, decorators: [{ type: Inject, args: [NGX_HIGHLIGHT_JS_URL,] }] }
    ]; };
    /** @nocollapse */ NgxHighlightJsThemeService.ngInjectableDef = defineInjectable({ factory: function NgxHighlightJsThemeService_Factory() { return new NgxHighlightJsThemeService(inject(NGX_HIGHLIGHT_JS_DEFAULT_THEME), inject(NGX_HIGHLIGHT_JS_URL)); }, token: NgxHighlightJsThemeService, providedIn: "root" });
    return NgxHighlightJsThemeService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxHighlightJsScriptService = /** @class */ (function () {
    function NgxHighlightJsScriptService(_jsURL) {
        this._jsURL = _jsURL;
        this._hljs = null;
        this._loadedLanguages = new Map();
    }
    /**
     * @return {?}
     */
    NgxHighlightJsScriptService.prototype.loaded = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this._loadedPromise) {
            return this._loadedPromise;
        }
        this._loadedPromise = new Promise(function (resolve, reject) {
            var /** @type {?} */ url = _this._jsURL + "/highlight.min.js";
            Scriptload.load(url)
                .then(function () {
                setTimeout(function () {
                    _this._hljs = window['hljs'];
                    resolve(_this._hljs);
                }, 10);
            })
                .catch(reject);
        });
        return this._loadedPromise;
    };
    /**
     * @param {?} lang
     * @return {?}
     */
    NgxHighlightJsScriptService.prototype.loadLanguage = /**
     * @param {?} lang
     * @return {?}
     */
    function (lang) {
        var /** @type {?} */ p = this._loadedLanguages.get(lang);
        if (p) {
            return p;
        }
        if (this._hljs && this._hljs.listLanguages().indexOf(lang) > -1) {
            p = Promise.resolve();
            this._loadedLanguages.set(lang, p);
            return p;
        }
        var /** @type {?} */ url = this._jsURL + "/languages/" + lang + ".min.js";
        p = this.loaded()
            .then(function () {
            return Scriptload.load(url);
        });
        this._loadedLanguages.set(lang, p);
        return p;
    };
    NgxHighlightJsScriptService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    NgxHighlightJsScriptService.ctorParameters = function () { return [
        { type: String, decorators: [{ type: Inject, args: [NGX_HIGHLIGHT_JS_URL,] }] }
    ]; };
    /** @nocollapse */ NgxHighlightJsScriptService.ngInjectableDef = defineInjectable({ factory: function NgxHighlightJsScriptService_Factory() { return new NgxHighlightJsScriptService(inject(NGX_HIGHLIGHT_JS_URL)); }, token: NgxHighlightJsScriptService, providedIn: "root" });
    return NgxHighlightJsScriptService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxHighlightJsService = /** @class */ (function () {
    function NgxHighlightJsService(themeService, scriptService) {
        this.themeService = themeService;
        this.scriptService = scriptService;
    }
    Object.defineProperty(NgxHighlightJsService.prototype, "theme", {
        get: /**
         * @return {?}
         */
        function () {
            return this.themeService._theme;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    NgxHighlightJsService.prototype.loaded = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this._loadedPromise) {
            return this._loadedPromise;
        }
        var /** @type {?} */ promises = [this.scriptService.loaded(), this.themeService.initialized()];
        this._loadedPromise = Promise.all(promises)
            .then(function (results) {
            _this._hljs = results[0];
            return (_this._hljs);
        });
        return this._loadedPromise;
    };
    /**
     * @param {?} lang
     * @return {?}
     */
    NgxHighlightJsService.prototype.loadLanguage = /**
     * @param {?} lang
     * @return {?}
     */
    function (lang) {
        return this.scriptService.loadLanguage(lang);
    };
    /**
     * @param {?} name
     * @return {?}
     */
    NgxHighlightJsService.prototype.loadTheme = /**
     * @param {?} name
     * @return {?}
     */
    function (name) {
        return this.themeService.loadTheme(name);
    };
    /**
     * @param {?} lang
     * @param {?} code
     * @return {?}
     */
    NgxHighlightJsService.prototype.highlight = /**
     * @param {?} lang
     * @param {?} code
     * @return {?}
     */
    function (lang, code) {
        var _this = this;
        return this.loaded()
            .then(function () {
            return _this.loadLanguage(lang);
        })
            .then(function () {
            var /** @type {?} */ hl = _this._hljs.highlight(lang, code, true).value;
            return hl;
        });
    };
    NgxHighlightJsService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    NgxHighlightJsService.ctorParameters = function () { return [
        { type: NgxHighlightJsThemeService },
        { type: NgxHighlightJsScriptService }
    ]; };
    /** @nocollapse */ NgxHighlightJsService.ngInjectableDef = defineInjectable({ factory: function NgxHighlightJsService_Factory() { return new NgxHighlightJsService(inject(NgxHighlightJsThemeService), inject(NgxHighlightJsScriptService)); }, token: NgxHighlightJsService, providedIn: "root" });
    return NgxHighlightJsService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxHighlightJsComponent = /** @class */ (function () {
    function NgxHighlightJsComponent(service, http) {
        this.service = service;
        this.http = http;
        this.highlightedHTML = '';
        this.error = null;
        this.afterInit = false;
    }
    /**
     * @return {?}
     */
    NgxHighlightJsComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.onChange();
        this.afterInit = true;
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    NgxHighlightJsComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (this.afterInit) {
            this.onChange();
        }
    };
    /**
     * @return {?}
     */
    NgxHighlightJsComponent.prototype.fetchText = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (!this.url) {
            return Promise.resolve(this.code || '');
        }
        return new Promise(function (resolve, reject) {
            _this.http.get(_this.url, { responseType: 'text' }).subscribe(function (text) { return resolve(text); }, function (err) { return reject(err); });
        });
    };
    /**
     * @return {?}
     */
    NgxHighlightJsComponent.prototype.onChange = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.error = null;
        this.isInitializing = true;
        this.fetchText()
            .then(function (text) {
            return _this.service.highlight(_this.lang, text);
        })
            .then(function (hl) {
            _this.highlightedHTML = hl;
            _this.isInitializing = false;
        })
            .catch(function (e) {
            _this.error = e.toString();
            _this.isInitializing = false;
        });
    };
    NgxHighlightJsComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-highlight-js',
                    exportAs: "ngxHighlightJs",
                    template: "<pre><code\n      [ngClass]=\"'hljs ' + lang\"\n      [innerHTML]=\"highlightedHTML\"></code></pre>",
                },] },
    ];
    /** @nocollapse */
    NgxHighlightJsComponent.ctorParameters = function () { return [
        { type: NgxHighlightJsService },
        { type: HttpClient }
    ]; };
    NgxHighlightJsComponent.propDecorators = {
        code: [{ type: Input }],
        url: [{ type: Input }],
        lang: [{ type: Input }]
    };
    return NgxHighlightJsComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxHighlightJsModule = /** @class */ (function () {
    function NgxHighlightJsModule() {
    }
    /**
     * @return {?}
     */
    NgxHighlightJsModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: NgxHighlightJsModule,
            providers: [
                NgxHighlightJsThemeService,
                NgxHighlightJsScriptService,
                NgxHighlightJsService,
                { provide: NGX_HIGHLIGHT_JS_URL, useValue: '//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0' },
                { provide: NGX_HIGHLIGHT_JS_DEFAULT_THEME, useValue: 'default' },
            ]
        };
    };
    NgxHighlightJsModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        HttpClientModule
                    ],
                    declarations: [NgxHighlightJsComponent],
                    exports: [NgxHighlightJsComponent],
                },] },
    ];
    return NgxHighlightJsModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NGX_HIGHLIGHT_JS_URL, NGX_HIGHLIGHT_JS_DEFAULT_THEME, NgxHighlightJsService, NgxHighlightJsModule, NgxHighlightJsScriptService as ɵb, NgxHighlightJsThemeService as ɵa, NgxHighlightJsComponent as ɵc };
